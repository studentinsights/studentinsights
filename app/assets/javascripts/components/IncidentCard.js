import React from 'react';


/*
DEPRECATED
Shows a "card" of an incident that happened with a student.
Note that this might contain sensitive data.
*/
class IncidentCard extends React.Component {
  render() {
    const {incident} = this.props;
    return (
      <div style={styles.box} key={incident.id}>
        <div style={styles.header}>
          <div style={styles.item}>
            <span style={styles.itemHead}>
              {'Date: '}
            </span>
            <span>
              {moment.utc(incident.occurred_at).format('MMM D, YYYY')}
            </span>
          </div>
          <div style={styles.centerItem}>
            <span style={styles.itemHead}>
              {'Code: '}
            </span>
            <span>
              {incident.incident_code}
            </span>
          </div>
          <div style={styles.item}>
            <span style={styles.itemHead}>
              {'Location: '}
            </span>
            <span>
              {incident.incident_location}
            </span>
          </div>
        </div>
        <div>
          <span style={styles.desc}>
            {'Description: '}
          </span>
        </div>
        <div>
          {incident.incident_description}
        </div>
      </div>
    );
  }
}
IncidentCard.propTypes = {
  incident: React.PropTypes.shape({
    id: React.PropTypes.number.isRequired,
    incident_code: React.PropTypes.string.isRequired,
    incident_location: React.PropTypes.string.isRequired,
    incident_description: React.PropTypes.string.isRequired
  })
};

const styles = {
  box: {
    border: '1px solid #ccc',
    padding:15,
    marginTop: 10,
    marginBottom: 10,
    width: '100%',
    backgroundColor: '#f2f2f2'
  },
  item: {
    paddingBottom: 10,
    width: 160
  },
  itemHead: {
    fontWeight: 'bold',
  },
  header: {
    display: 'flex',
    flexFlow: 'row',
    justifyContent: 'space-between'
  },
  desc: {
    fontWeight: 'bold',
    paddingTop: 30
  },
  centerItem: {
    paddingBottom: 10,
    textAlign: 'center',
    width: 75
  }
};

export default IncidentCard;