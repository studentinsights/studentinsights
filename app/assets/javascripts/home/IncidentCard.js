import React from 'react';
import {toMomentFromTime} from '../helpers/toMoment';
import FeedCardFrame from './FeedCardFrame';
import HouseBadge from '../components/HouseBadge';
import Badge from '../components/Badge';


// Render a card in the feed for a discipline incident
class IncidentCard extends React.Component {
  render() {
    const {nowFn} = this.context;
    const now = nowFn();
    const {style, incidentCard} = this.props;
    const {student} = incidentCard;

    return (
      <div className="IncidentCard" >
        <FeedCardFrame
          style={style}
          student={student}
          whereEl={<div>in {incidentCard.incident_location}</div>}
          whenEl={<div>{toMomentFromTime(incidentCard.occurred_at).from(now)} on {toMomentFromTime(incidentCard.occurred_at).format('M/D')}</div>}
          badgesEl={<div>
            {student.house && <HouseBadge style={styles.footerBadge} house={student.house} />}
            <Badge style={styles.footerBadge} text="Incident" backgroundColor="rgb(255, 140, 0)" />
          </div>}
        >
          <div>{incidentCard.incident_description} (code: {incidentCard.incident_code})</div>
        </FeedCardFrame>
      </div>
    );
  }
}
IncidentCard.contextTypes = {
  nowFn: React.PropTypes.func.isRequired
};
IncidentCard.propTypes = {
  incidentCard: React.PropTypes.shape({
    id: React.PropTypes.number.isRequired,
    incident_code: React.PropTypes.string.isRequired,
    incident_location: React.PropTypes.string.isRequired,
    incident_description: React.PropTypes.string.isRequired,
    occurred_at: React.PropTypes.string.isRequired,
    has_exact_time: React.PropTypes.bool.isRequired,
    student: React.PropTypes.shape({
      id: React.PropTypes.number.isRequired,
      first_name: React.PropTypes.string.isRequired,
      last_name: React.PropTypes.string.isRequired,
      grade: React.PropTypes.string.isRequired,
      house: React.PropTypes.string,
      homeroom: React.PropTypes.shape({
        id: React.PropTypes.number.isRequired,
        name: React.PropTypes.string.isRequired,
        educator: React.PropTypes.object
      })
    })
  }),
  style: React.PropTypes.object
};


const styles = {
  footerBadge: {
    marginLeft: 5
  }
};

export default IncidentCard;