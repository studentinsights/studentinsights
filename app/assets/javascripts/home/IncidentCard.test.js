import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import IncidentCard from './IncidentCard';
import {withDefaultNowContext} from '../../../../spec/javascripts/support/NowContainer';


function testProps(props = {}) {
  return {
    incidentCard: {
      "id": 21,
      "incident_code": "OT",
      "incident_location": "Locker room/gym",
      "incident_description": "Earlier today something happened with the student that was concerning and we did this to de-escalate and support them in the moment, and then took these steps to understand the underlying challenges they're facing, and adapt how we're working with them moving forward.",
      "occurred_at": "2018-02-28T05:00:00.000Z",
      "has_exact_time": false,
      "student": {
        "id": 100,
        "grade": "10",
        "first_name": "Mowgli",
        "last_name": "Pan",
        "house": "Beacon",
        "homeroom": {
          "id": 4,
          "name": "SHS ALL"
        }
      }
    },
    ...props
  };
}

it('renders without crashing', () => {
  const props = testProps();
  const el = document.createElement('div');
  ReactDOM.render(withDefaultNowContext(<IncidentCard {...props} />), el);
});

it('matches snapshot', () => {
  const props = testProps();
  const tree = renderer
    .create(withDefaultNowContext(<IncidentCard {...props} />))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
