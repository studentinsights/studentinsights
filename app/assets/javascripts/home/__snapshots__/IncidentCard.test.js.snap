// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="IncidentCard"
>
  <div
    className="Card FeedCardFrame"
    style={
      Object {
        "border": "1px solid #ccc",
        "borderRadius": 3,
        "padding": 10,
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <div>
          <div>
            <a
              href="/students/100"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Mowgli
               
              Pan
            </a>
          </div>
          <div>
            10th grade
          </div>
          <div>
            <span
              className="Homeroom"
              style={Object {}}
            >
              <a
                href="/homerooms/4"
              >
                SHS ALL
              </a>
            </span>
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <div />
        <div>
          <div>
            in 
            Locker room/gym
          </div>
        </div>
        <div>
          <div>
            13 days ago
             on 
            2/28
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "marginBottom": 20,
          "marginTop": 20,
        }
      }
    >
      <div>
        Earlier today something happened with the student that was concerning and we did this to de-escalate and support them in the moment, and then took these steps to understand the underlying challenges they're facing, and adapt how we're working with them moving forward.
         (code: 
        OT
        )
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "flex-end",
          "marginBottom": 5,
        }
      }
    >
      <div>
        <span
          className="HouseBadge"
          style={
            Object {
              "backgroundColor": "#333",
              "color": "white",
              "marginLeft": 5,
              "opacity": 0.5,
              "padding": 5,
            }
          }
        >
          Beacon house
        </span>
        <span
          className="Badge"
          style={
            Object {
              "backgroundColor": "rgb(255, 140, 0)",
              "color": "white",
              "marginLeft": 5,
              "opacity": 0.5,
              "padding": 5,
            }
          }
        >
          Incident
        </span>
      </div>
    </div>
  </div>
</div>
`;
